java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -137,6 +139,10 @@ public boolean isNull(ProgramState ps, SymbolicValue val) {
     return NullConstraint.NULL.equals(ps.constraints.get(val));
   }
 
+  public boolean isConstrained(ProgramState ps, SymbolicValue val) {
+    return ps.constraints.containsKey(val);
+  }
+
   public Pair<ProgramState, ProgramState> assumeDual(ProgramState programState, Tree condition) {
     // FIXME condition value should be evaluated to determine if it is worth exploring this branch. This should probably be done in a
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1134)